src/Maybe.js

```diff
@@ -1,45 +1,82 @@
-module.exports = Maybe;
+var util = require('./internal/util.js');
 
 function Maybe(x) {
-    if (!(this instanceof Maybe)) {
-        return new Maybe(x);
-    }
+    throw new Error('Maybe cannot be directly instantiated.');
```

Comment:
@buzzdecafe Agreed, but my question here was about the constructor? Why throw an error instead of doing the null check? The constructor is not a part of the specification right?